File: string/decrypt-string-from-alphabet-to-integer-mapping/decrypt-string-from-alphabet-to-integer-mapping.ts
function freqAlphabets(s: string): string {
    let res = '';
    for (let i = 0; i < s.length;) {
        if (i + 2 < s.length && s[i + 2] === "#") {
            res += String.fromCharCode('j'.charCodeAt(0) +
                (s[i].charCodeAt(0) - '1'.charCodeAt(0)) * 10 +
                s[i + 1].charCodeAt(0) - '0'.charCodeAt(0)
            );
            i += 3;
        } else {
            res += String.fromCharCode('a'.charCodeAt(0)
                + s[i].charCodeAt(0) - '1'.charCodeAt(0))
            i++;
        }

    }

    return res;
};

File: string/decrypt-string-from-alphabet-to-integer-mapping/decrypt-string-from-alphabet-to-integer-mapping.go
package decryptstringfromalphabettointegermapping

func freqAlphabets(s string) string {
	out := []byte{}
	for i := 0; i < len(s); {
		if i+2 < len(s) && s[i+2] == '#' {
			out = append(out, 'j'+(s[i]-'1')*10+s[i+1]-'0')
			i += 3
		} else {
			out = append(out, 'a'+s[i]-'1')
			i++
		}
	}

	return string(out)
}
